wazo_agid/objects.py

```diff
@@ -906,33 +906,44 @@ def __init__(self, agi, cursor, context):
 class CallerID(object):
     @staticmethod
     def parse(callerid):
+        logger.debug('caller_id parse: parsing "%s"', callerid)
         m = CALLERID_MATCHER(callerid)
 
         if not m:
+            logger.debug('caller_id parse: could not match callerid, giving up')
             return
 
         calleridname = m.group(1)
         calleridnum = m.group(3)
+        logger.debug('caller_id parse: calleridname: "%s", calleridnum: "%s"', calleridname, calleridnum)
 
         if calleridname is None:
             calleridname = m.group(2)
+            logger.debug('caller_id parse: calleridname: "%s", calleridnum: "%s"', calleridname, calleridnum)
 
             if calleridnum is None and CALLERIDNUM_MATCHER(calleridname):
                 calleridnum = m.group(2)
+                logger.debug('caller_id parse: calleridname: "%s", calleridnum: "%s"', calleridname, calleridnum)
 
+        logger.debug('caller_id parse: done')
         return (calleridname, calleridnum)
 
     @staticmethod
     def set(agi, callerid):
+        logger.debug('caller_id set: parsing "%s"', callerid)
         cid_parsed = CallerID.parse(callerid)
+        logger.debug('caller_id set: parsed caller id to "%s"', cid_parsed)
 
         if not cid_parsed:
+            logger.debug('caller_id set: could not parse caller id, giving up')
             return
 
         calleridname, calleridnum = cid_parsed
+        logger.debug('caller_id set: calleridname: "%s", calleridnum: "%s"', calleridname, calleridnum)
 
         if calleridname is None and calleridnum is not None:
             calleridname = calleridnum
+            logger.debug('caller_id set: calleridname: "%s", calleridnum: "%s"', calleridname, calleridnum)
```

Comment:
Maybe we can merge these log together at the end of function? Otherwise we cannot distinguish their meaning

Reply:
Same as above: we want to know exactly which code is executed. I added a description in the log message to distinguish them.